libs/membership/api/jest.config.ts

```diff
@@ -13,5 +13,6 @@ export default {
   },
   moduleFileExtensions: ['ts', 'js', 'html'],
   coverageDirectory: '../../../coverage/libs/membership/api',
-  globalSetup: '<rootDir>/setup-database.js'
+  globalSetup: '<rootDir>/setup-database.js',
+  maxWorkers: 1
```

Comment:
No, they don't depend on other tests, but running multiple tests in parallel breaks them, because they access the same database.

Reply:
If the tests clear the whole DB, they do rely on other tests not running, so there is a dependency. maxWorkers shouldn't be set to 1 (or use run in band). They should be able to run in parallel or independently.

I would rather not introduce this, as this will lower the quality of our tests  and in the future.

Reply:
@Itrulia @tomaszdurka 

> I would rather not introduce this, as this will lower the quality of our tests and in the future.

I understand, but we didn't introduce this. The problem already existed for example in [DashboardSubscriptionResolver](https://github.com/wepublish/wepublish/blob/f/membership-config/libs/membership/api/src/lib/dashboard/subscriptions/dashboard-subscription.resolver.spec.ts#L92-L94), where the beforeEach call clears all Subscriptions from the database. If this line is run while another test checks for the count or existence of subscriptions, it will fail. We only provided a solution to the existing problem by running the tests concurrently.

One alternative would be to connect to a different copy of the database for each jest worker thread, so they don't interfere. According to the jest issue tracker, [this is not easily doable](https://github.com/jestjs/jest/issues/8708).

Mocking the Prisma client would be possible, but we would lose quite a bit of testability there, because for example testing if records are returned in a specific order would no longer be possible (or depend entirely on the mock).